### DISCUSSION TITLE
Support for Jsonnet

### DISCUSSION THREAD

**Discussion by netlander** [woodpecker-ci/woodpecker#3277] [Request a Feature] [answered]
Curently using Drone CI and would be interested in switching to Woodpecker. However, all our build and release scripts are in `jsonnet` format and it would be impossible to migrate without that support.

Are ther any plans on the horizon for `jsonnet`?

**Comment by qwerty287** [comment, marked as answer]
Please search open issues and discussions before opening new ones.

https://github.com/woodpecker-ci/woodpecker/discussions/2254 will cover this issue. You can currently use addons to make this working already: https://github.com/woodpecker-ci/woodpecker/discussions/2254#discussioncomment-8171108

**Reply by 6543** [reply]
( https://codeberg.org/qwerty287/woodpecker-jsonnet-addon )

**Comment by netlander** [comment]
I think I've seen the discussion before posting, the impression I had was: this solution is half baked esp. if it doesn't support private repos. The thing about "need to compile yourself" suggests the same.

![image](https://github.com/woodpecker-ci/woodpecker/assets/4721044/3401046d-da8a-4c7b-b796-cc457aaa448f)

We don't have Go development talent on board and hoped that a solution like the above would be implemented in a straightforwad manner.

This is the reason that I've started this conversation, feel free to close if you think this is not a conversation worth having.

**Reply by anbraten** [reply]
Actually looking at this I think it might help to do this via the config api instead of the plugin approach. That way the admin would just need to add an additional container to the docker-compose and set the config-api-url. The config api also provides the netrc credentials, so we could check for the corresponding config files. @qwerty287 What do you think? Should I give it a try or should we do it using #3268.

**Reply by qwerty287** [reply]
@anbraten 

> That way the admin would just need to add an additional container to the docker-compose and set the config-api-url.

I didn't write the addon as a complete, working solution, but as an example you should adapt to your needs.

If something like this is released as "ready to use", it should be possible to download a binary and update the addon env var, this does not require more effort.

> The config api also provides the netrc credentials, so we could check for the corresponding config files

The addon gets them as well, but netrc data is different for every forge. E.g. bitbucket sets the `login` parameter` as token, while others use the specific `token` setting (not sure if this is really how it looks, but it's in this direction).

> Should I give it a try or should we do it using #3268.

Feel free to do it, but I'm also fine with doing in #3268. In general, I think we should rather use addons instead of http services. With #3268, they can also be written in any language, go experts are not needed anymore.

To fix the token issue, I plan the following: The addons get the way to perform an RPC call to the server to get the user token of a repo. Maybe also some way to allow using all forge functions.

@netlander 

> this solution is half baked esp. if it doesn't support private repos

Yes, it is not complete. It was more meant as example you can use to adapt it to your needs.

> We don't have Go development talent on board

There is also a comment in the linked discussion about this: You can also use http config services for now. Addons need a big rework - see #3268 - and when I finish this PR, it will be possible to write addons in any language.
